src/integrations/admin/link-count-tools-integration.php

```diff
@@ -0,0 +1,171 @@
+use Yoast\WP\SEO\Routes\Link_Indexing_Route;
+
+/**
+ * Link_Count_Tools_Integration class
+ */
+class Link_Count_Tools_Integration implements Integration_Interface {
+
+	/**
+	 * @inheritDoc
+	 */
+	public static function get_conditionals() {
+		return [
+			Admin_Conditional::class,
+			Yoast_Admin_And_Dashboard_Conditional::class,
+			Migrations_Conditional::class,
+		];
+	}
+
+	/**
+	 * The post link indexing action.
+	 *
+	 * @var Post_Link_Indexing_Action
+	 */
+	protected $post_link_indexing_action;
+
+	/**
+	 * The term link indexing action.
+	 *
+	 * @var Term_Link_Indexing_Action
+	 */
+	protected $term_link_indexing_action;
+
+	/**
+	 * Represents the admin asset manager.
+	 *
+	 * @var WPSEO_Admin_Asset_Manager
+	 */
+	protected $asset_manager;
+
+	/**
+	 * The total amount of unindexed objects.
+	 *
+	 * @var int
+	 */
+	private $total_unindexed;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Post_Link_Indexing_Action $post_link_indexing_action The post link indexing action.
+	 * @param Term_Link_Indexing_Action $term_link_indexing_action The term link indexing action.
+	 * @param WPSEO_Admin_Asset_Manager $asset_manager             The asset manager.
+	 */
+	public function __construct(
+		Post_Link_Indexing_Action $post_link_indexing_action,
+		Term_Link_Indexing_Action $term_link_indexing_action,
+		WPSEO_Admin_Asset_Manager $asset_manager
+	) {
+		$this->post_link_indexing_action = $post_link_indexing_action;
+		$this->term_link_indexing_action = $term_link_indexing_action;
+		$this->asset_manager             = $asset_manager;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function register_hooks() {
+		\add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_assets' ], 10 );
+		\add_action( 'wpseo_tools_overview_list_items', [ $this, 'render_tools_overview_item' ], 10 );
+	}
+
+	/**
+	 * Enqueues all required assets.
+	 *
+	 * @return void
+	 */
+	public function enqueue_assets() {
+		if ( $this->get_total_unindexed() === 0 ) {
+			return;
+		}
+
+		\add_action( 'admin_footer', [ $this, 'render_modal' ], 20 );
+
+		$this->asset_manager->enqueue_script( 'indexation' );
+		$this->asset_manager->enqueue_style( 'admin-css' );
+
+		$data = [
+			'amount'  => $this->get_total_unindexed(),
+			'ids'     => [
+				'modal'        => '#yoast-link-indexing-modal',
+				'count'        => '#yoast-link-indexing-current-count',
+				'progress'     => '#yoast-link-indexing-progress-bar',
+				'tool'         => '#yoast-link-indexing',
+				'notification' => '#yoast-link-indexing-notification',
+			],
+			'restApi' => [
+				'root'      => \esc_url_raw( \rest_url() ),
+				'endpoints' => [
+					'posts' => Link_Indexing_Route::FULL_POSTS_ROUTE,
+					'terms' => Link_Indexing_Route::FULL_TERMS_ROUTE,
+				],
+				'nonce'     => \wp_create_nonce( 'wp_rest' ),
+			],
+			'message' => [
+				'indexingCompleted' => '<span class="yoast-check">' . \esc_html__( 'Good job! All the links in your texts have been counted.', 'wordpress-seo' ) . '</span>',
```

Comment:
Any reason why these translations are not JS territory?

Reply:
This same settings object is passed for creating indexables, counting links and gathering prominent words. Having these here avoids out-of-context switch statements in the generic Javascript that handles all 3 of these.